1-Draft/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,473 @@
+
+```output
+Version  Name       Type    Repository  Description
+-------  ----       ----    ----------  -----------
+5.0.0    VSTeam     Module  PSGallery   Adds functionality for working with Visual …
+0.14.94  PSGitHub   Module  PSGallery   This PowerShell module enables integration …
+0.7.3    posh-git   Module  PSGallery   Provides prompt with Git status summary …
+1.0.1    PSAutoMute Script  PSGallery   Powershell script to Auto Mute you sound devices …
+```
+
+### Uninstalling resources
+
+`Uninstall-PSResource` will be available to remove installed resources.
+
+>[!NOTE]
+>Uninstalling dependencies automatically will be something to consider in the future.
+
+If a package is unable to be uninstalled because it is currently loaded, the cmdlet will emit a warning
+which enumerates the packages that were unable to be installed.
+We will explore a dependency clean parameter in a future version of PowerShellGet.
+
+When uninstalling the dependency check will be performed not only on the RequiredModules.Name,
+but also on RequiredModules.Version, which should be less-than-or-equal to the version of the module being
+uninstalled in order to be considered a dependency.
+
+### Proxy support
+
+Each cmdlet will have `-Proxy` and `-ProxyCredential` parameters.
+
+### PowerShellGallery status
+
+Upon failure to connect to PSGallery, the cmdlets should retrieve status from
+a well known REST API and return a more descriptive error message to the user.
+
+### Open Sourcing the module
+
+In order to open source PowerShellGet 3.0 we will fork the current PowerShellGet repo, rename it as PowerShellGet2,
+and use the existing PowerShellGet repo as the primary support and development contact for PowerShellGet 3+.
+
+## Alternate Proposals and Considerations
+
+These are items are outside the scope of this RFC and version 3.0.
+Many of these items can be addressed in a future version of PowerShellGet 3.x without
+introducing a breaking change:
+
+- This RFC does not cover the module authoring experience on publishing a cross-platform
+  module with multiple dependencies and supporting multiple runtimes.
+
+- [Should be the opposite] If there is a desire to explicitly update the local cache (like `apt`), we can introduce a
+  `Update-PSResourceCache` cmdlet in the future and a property on PSRepository registrations
+  indicating whether it auto-updates or not.
+
+- The ability to set policy for PSGet
+
+- Automatic cleanup of old resources when upgrading or removing resources
+
+- Signing incorporated into `Publish-PSResource`
+
+- `-Trusted` switch to `Install-PSResource` and `Find-PSResource` that pre-validates
+  the signing cert is trusted on the system
+
+- Notify users that they have outdated versions of a module (just patch versions?)
+
+- A way to filter resources to those allowed to be installed on the system
+
+- Enterprise management of local cache/the ability to configure a PSRepository on a system-wide level

```

Comment:
Is this list set in a priority order? Because having the possibility to set a repository system-wide is really some **strongly needed** for us. 
We would benfit way more from having a repository system wide, than from having a message saying we have don't have the latest module installed, or from a cleanup of old modules (which we intentionnally dont remove for backwards compatability / fallback scenarios).

Reply:
Agreed, and for v3 GA this is a critical need to address the long standing issue that v2 has had requiring per user setup which doesn't work for systems that require shared management at all